napari/layers/utils/layer_utils.py

```diff
@@ -204,81 +216,485 @@ def _nanmax(array):
     return max_value
 
 
-def calc_data_range(data, rgb=False) -> Tuple[float, float]:
+def calc_data_range(data, rgb: bool = False) -> None | Tuple[float, float]:
     """Calculate range of data values. If all values are equal return [0, 1].
 
     Parameters
     ----------
     data : array
         Data to calculate range of values over.
     rgb : bool
-        Flag if data is rgb.
+        Flag if data is rgb. If so, return [0, 255].
 
     Returns
     -------
-    values : pair of floats
-        Minimum and maximum values in that order.
+    values : None | tuple[float, float]
+        Minimum and maximum values in that order or None if dtype is float.
 
     Notes
     -----
-    If the data type is uint8, no calculation is performed, and 0-255 is
+    If the data type is uint8 or rgb, no calculation is performed, and 0-255 is
     returned.
     """
-    if data.dtype == np.uint8:
-        return (0, 255)
-
-    center: Union[int, List[int]]
-
-    if data.size > 1e7 and (data.ndim == 1 or (rgb and data.ndim == 2)):
-        # If data is very large take the average of start, middle and end.
-        center = int(data.shape[0] // 2)
-        slices = [
-            slice(0, 4096),
-            slice(center - 2048, center + 2048),
-            slice(-4096, None),
-        ]
+    dtype = normalize_dtype(getattr(data, 'dtype', None))
+    # Vispy only supports uint8 RGB for which we set these clims
+    if rgb and dtype != np.uint8:
+        # Vispy does not display rgb other than uint8, so it casts it to uint8
+        show_info(
```

Comment:
```suggestion
        warnings.warns(
```

I show_info requires napari viewer, when this code may be caused when contruct layer, before viewer creation.